Tests/TuistKitTests/CommandTracking/TrackableCommandTests.swift

```diff
@@ -8,66 +10,55 @@ import XCTest
 
 final class TrackableCommandTests: TuistUnitTestCase {
     private var subject: TrackableCommand!
-    private var mockCommandEventTagger: MockCommandEventTagging!
+    private var mockAsyncQueue: MockAsyncQueuer!
 
     override func setUp() {
         super.setUp()
-        mockCommandEventTagger = MockCommandEventTagging()
+        mockAsyncQueue = MockAsyncQueuer()
     }
 
     override func tearDown() {
         subject = nil
-        mockCommandEventTagger = nil
+        mockAsyncQueue = nil
         super.tearDown()
     }
 
     private func makeSubject(flag: Bool = true) {
         subject = TrackableCommand(command: TestCommand(flag: flag),
-                                   commandEventTagger: mockCommandEventTagger,
-                                   clock: WallClock())
+                                   clock: WallClock(),
+                                   asyncQueue: mockAsyncQueue)
     }
 
     // MARK: - Tests
 
-    func test_whenFlagTrue_callsCommandEventTaggerWithExpectedParameters() throws {
+    func test_whenParamsHaveFlagTrue_dispatchesEventWithExpectedParameters() throws {
         // Given
         makeSubject(flag: true)
         let expectedParams = ["flag": "true"]
 
         // When
-        try subject.run()
+        try subject.run(completion: {})
```

Comment:
Could we test that the given completion block is called?

Reply:
Yep, updated!